addons/web/static/tests/webclient/commands/command_palette_tests.js

```diff
@@ -366,6 +367,46 @@ QUnit.test("multi provider with the same namespace", async (assert) => {
     );
 });
 
+QUnit.test("check the concurrency during a research", async (assert) => {
+    testComponent = await mount(TestComponent, { env, target });
+    const imSearchDef = makeDeferred();
+    const provide = async (env, options) => {
+        if (options.searchValue) {
+            await imSearchDef;
+        }
+        return [
+            {
+                name: "a",
+                action: () => {
+                    assert.step("a");
+                },
+            },
+            {
+                name: "b",
+                action: () => {
+                    assert.step("b");
+                },
+            },
+        ];
+    };
+    const providers = [{ namespace: "default", provide }];
+    const config = {
+        providers,
+    };
+    env.services.dialog.add(CommandPaletteDialog, {
+        config,
+    });
+    await nextTick();
+    assert.containsOnce(target, ".o_command_palette");
+    assert.containsN(target, ".o_command", 2);
+
+    await editSearchBar("b");
+    triggerHotkey("enter");
+    imSearchDef.resolve();
+    await nextTick();
+    assert.verifySteps(["b"]);
```

Comment:
I don't understand the test, ite seems wrong

I would have add an await nextTick() after the triggerHotkey and an empty verifySteps([]) before the imSearch.resolve()